src/field/gf_101_2.rs

```diff
@@ -0,0 +1,335 @@
+use core::{
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Mul, Neg, Sub, SubAssign},
+};
+use std::ops::{Div, DivAssign, MulAssign, Rem};
+
+use serde::{Deserialize, Serialize};
+
+use super::ExtensionField;
+use crate::field::FiniteField;
+
+/// Pluto curve with modulus 101 supports two degree extension field. This can be verified
+/// by finding out embedding degree of the curve, i.e. smallest k: r|q^k-1
```

Comment:
When reading about the embedding degree I was trying to see this by hand but came across some questions. In our case r is our curve order which is 17 and p is field order which is 101. How does 17 | 101 ^2 - 1 which is 17 | 10200. 17 doesn't divide 10200. Also 10200 mod 101 is just 17 | 100. Do I have an error in my arithmetic? Are we working in a field here?

Reply:
that's a classic, my bad.